examples/provider/provider.tf

```diff
@@ -233,5 +314,22 @@ resource "dcloud_ip_nat_rule" "ip_nat_rule"{
 resource "dcloud_vm_nat_rule" "vm_nat_rule"{
   topology_uid = dcloud_topology.test_topology.id
   nic_uid = dcloud_vm.vm1.network_interfaces[1].uid
-  east_west = false
+  east_west = true
+}
+
+resource "dcloud_inbound_proxy_rule" "inbound_proxy_rule"{
+  topology_uid = dcloud_topology.test_topology.id
+  nic_uid = dcloud_vm.vm2.network_interfaces[1].uid
+  target_vm_name = dcloud_vm.vm1.name
+  tcp_port = 443
+  url_path = "/testing/url/"
+  hyperlink = "Test Hyperlink"
+  ssl = true
+  show_hyperlink = true
+}
+
+resource "dcloud_mail_server" "mail_server"{
+  topology_uid = dcloud_topology.test_topology.id
+  nic_uid = dcloud_vm.vm4.network_interfaces[0].uid
+  dns_asset_id = "3"
```

Comment:
Does this imply the need for a Inventory Data Source for DNS Assets?

Reply:
Yes Hugh. I have it done in the DNS feature, and it will be there in the next PR.